src/zenml/services/service.py

```diff
@@ -42,15 +42,51 @@ class ServiceConfig(BaseTypedModel):
         name: name for the service instance
         description: description of the service
         pipeline_name: name of the pipeline that spun up the service
-        pipeline_run_id: ID of the pipeline run that spun up the service
         pipeline_step_name: name of the pipeline step that spun up the service
+        run_name: name of the pipeline run that spun up the service.
     """
 
     name: str = ""
     description: str = ""
     pipeline_name: str = ""
-    pipeline_run_id: str = ""
     pipeline_step_name: str = ""
+    run_name: str = ""
+
+    @property
+    def pipeline_run_id(self) -> str:
+        """Getter for the pipeline_run_id attribute.
+
+        Returns:
+            The pipeline_run_id attribute.
+        """
+        logger.warning(
+            "The 'pipeline_run_id' attribute is deprecated. Use 'run_name' instead.",
+            DeprecationWarning,
+        )
+        return self._pipeline_run_id
+
+    @pipeline_run_id.setter
+    def pipeline_run_id(self, value: str) -> None:
+        """Setter for the pipeline_run_id attribute."""
+        logger.warning(
+            "The 'pipeline_run_id' attribute is deprecated. Use 'run_name' instead.",
+            DeprecationWarning,
+        )
+        self._pipeline_run_id = value
+
+    def __setattr__(self, name: str, value: str) -> None:
+        """Setter for the run_name attribute.
+
+        If the run_name attribute is set to an empty string, it will be
+        automatically set to the pipeline_run_id value.
+
+        Args:
+            name: name of the attribute.
+            value: value of the attribute.
+        """
+        if name == "run_name" and not value:
```

Comment:
Can you explain when this case would happen, someone setting the run name to `None`/empty string but still setting the old `pipeline_run_id`?

Reply:
i thought maybe, if we try to load back a service of a model that was already deployed